### PR TITLE
Make slices.Map UnmarshalJSON a pointer receiver

### PR DESCRIPTION
**Pull request by naemono**
Fixes https://github.com/gobuffalo/pop/issues/654

### CODE REVIEW

**Comment by aeneasr** [comment]
Could you add a test please? Thank you!

**Comment by qindj** [comment]
I have the same issue, value will be nil when slices.map used as a table's column type

File: slices/map.go [hunk 1]
```diff
@@ -47,17 +47,17 @@ func (m Map) Value() (driver.Value, error) {
 
 // UnmarshalJSON will unmarshall JSON value into
 // the map representation of this value.
-func (m Map) UnmarshalJSON(b []byte) error {
+func (m *Map) UnmarshalJSON(b []byte) error {
```
Comment (sio4) [inline slices/map.go:50]: It should be! Thanks for correcting the code!

**Review by sio4 — APPROVED** [review]

**Comment by sio4** [comment]
I will cover the test cases myself. Thanks!

**Pull request merged**